### DISCUSSION TITLE
HowTo resolve: Property 'updatedAt' is missing in type....?

### DISCUSSION THREAD

**Discussion by TheoMer** [prisma/prisma#3178] [Q&A] [answered]
So I have an Item model defined as follows:

```prisma
model Item {
  ...
  createdAt       DateTime       @default(now())
  updatedAt       DateTime
  itemvariants    ItemVariants[]
}

model ItemVariants {
  ...
  createdAt       DateTime   @default(now())
  itemId          String
  updatedAt       DateTime
  item            Item      @relation(fields: [itemId], references: [id])
}
```

which I define in graphql.ts as:

```ts
  schema.objectType({
    name: 'Item',
    definition(t) {
      t.model.itemvariants({
        filtering: true,
        ordering: true,
      })
      t.date("createdAt")
      t.date("updatedAt")
    },
  })

  schema.objectType({
    name: 'ItemVariants',
    definition(t) {
      t.model.itemId()
      t.model.item()
      t.date("createdAt")
      t.date("updatedAt")
    },
  })
```

The error is being issued because in my createItem I have not specified createdAt/updatedAt attributes. How do I do this?

```
      t.field('createItem', {
        type: "Item",
        nullable: false,
        args: {
          ...
         itemvariants: schema.arg({ type: "ItemVariantsCreateManyWithoutItemInput", list: true, nullable: false }),
         updatedAt: ???????
         createdAt: ???????
        },
        resolve: async (_, args, ctx) => {
          return await ctx.db.item.create(
            {
              data: {
                ...args,
                user: {
                  connect: {
                    id: ctx.req.userId,
                  },
                },
              },
            },
          )
        }
      })
```

**Comment by ryands17** [comment, marked as answer]
Hey @TheoMer 👋 
I think the issue is because of `updatedAt` and not `createdAt`. `createdAt` already has a default value so it is not required to be passed while creating, but you would need to add this to the schema for automatic insertion of updated time.

```prisma
model Item {
  createdAt       DateTime       @default(now())
  updatedAt       DateTime      @updatedAt
  itemvariants    ItemVariants[]
}
```

**Comment by TheoMer** [comment]
Hey @ryands17 

I made the amendment as recommended

```
updatedAt       DateTime      @updatedAt
```

but VSCode is still issuing the following error:

```
The data needed to create a Item.

Property 'updatedAt' is missing in type '{ user: { connect: { id: string; }; }; color: { connect?: { id?: string | null | undefined; name?: string | null | undefined; } | null | undefined; create?: { createdAt?: any; id: string; ItemVariants?: { ...; } | ... 1 more ... | undefined; label?: string | ... 1 more ... | undefined; name: string; OrderItem?: { .....' but required in type 'ItemCreateInput'.ts(2741)
```

**Reply by ryands17** [reply]
Did you run `prisma generate` and the Nexus schema has updated as well? If it still doesn't work, restart the editor and check again.

**Reply by TheoMer** [reply]
Hey @ryands17 that did the trick. many thanks.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
